### ISSUE TITLE
Back-Channel Logout

### ISSUE THREAD

**Issue by yuriyz** [GluuFederation/oxAuth#70]
Add support for back-channel logout 
http://openid.net/specs/openid-connect-backchannel-1_0.html#BCRequest

**Comment by yurem** [comment]
You know more in this area. Can you explain or give us example where we can use back channel logout. Can we just call https://server/oxauth/seam/resource/restv1/oxauth/end_session to end session?

**Comment by nynymike** [comment]
I think this method requires the RP to have an endpoint to receive a notification from the OP that a logout has occurred. Originally, Google was dead set against this. They can't support this because its too hard to remember all the clients that require a logout notification, and it would have a big network impact. However, I think that many enterprises have the requirement, and its was decided that OpenID Connect needed this if it was going to replace SAML.

**Comment by yurem** [comment]
Also I see that RP should issue additional token to conform this draft spec. This require additional resources..
In oxTust we implemented something similar. We call end_session directly (without user browser) on oxTrust session destroy: https://github.com/GluuFederation/oxTrust/blob/master/server/src/main/java/org/gluu/oxtrust/service/AuthenticationSessionService.java#L39

**Comment by yuriyz** [comment]
Ok, we have to differ 3 ways of logout:
1. end session according to http://openid.net/specs/openid-connect-session-1_0.html#OPMetadata
2. http based logout http://openid.net/specs/openid-connect-logout-1_0.html
3. back channel logout http://openid.net/specs/openid-connect-backchannel-1_0.html

@yurem yes, end_session is enough for oxTrust, so we don't need to change anything there.

From spec:
```
end_session - URL at the OP to which an RP can perform a redirect to request that the End-User be logged out at the OP. 
```

About back channel logout, to be honest I have to dig deeper to say more about pros&cons but at first glance it looks reasonable.

**Comment by yurem** [comment]
@yuriyz , you paste right definition of end_session endpoint.
But our end_session endpoint implementation allows to call it in similar to back channel way.
RP can call directly OP end_session endpoint without user browser.
RP only need to store in session id_token once user login.

For me it's much easier now to use end_session than implement Draft 01 version of this spec. I believe final version or at least Draft v10 will be changed by 50% percent.

Let me sum up. I not offer to forget about this spec. We need to support more spec to make better server. I offer to wait for more stable spec version and use our end_session extension in similar to back channel way.

**Comment by nynymike** [comment]
Here is an article from Vlad on back channel logout http://gluu.co/back-logout-vlad

**Comment by yuriyz** [comment]
Backchannel logout is implemented in 4.2.
